handbook.md

```diff
@@ -0,0 +1,256 @@
+---
+layout: contribute
+title: Handbook
+---
+
+# How we make things
+
+## Branching and SCM
+
+We use git for source control. We follow this model in what we call the core projects ([Foreman](https://github.com/theforeman/foreman), [Smart Proxy](https://github.com/theforeman/smart-proxy), [Foreman installer](https://github.com/theforeman/foreman-installer) and [Foreman SELinux](https://github.com/foreman-selinux)) for branching:
+
+* **develop**
+    * New pull requests should be made against this branch.
+    * All pull requests are merged here.
+    * Contains nightlies code. It's relatively stable, but we make no promises about it.
+    * [`test_develop` in Jenkins](http://ci.theforeman.org/job/test_develop/) ensures our tests are passing at all times for various configurations.
+* **release-stable (i.e. 1.10-stable)** -
+    * Individual developers do not need to maintain this, it's part of our [release process](http://projects.theforeman.org/projects/foreman/wiki/Release_Process#Preparing-a-release)
+    * Code is cherry-picked using `git cherry-pick -x` into this branch.
+    * It contains the latest stable code for a release cycle, 1.8-stable contains 1.8.1, 1.8.2, 1.8.3...
+    * Typically, we will announce the start of a new branch
+
+And we tag releases:
+
+* **release (i.e. 1.10)**
+    * Code is cherry-picked using `git cherry-pick -x` into this tag.
+    * It contains the latest stable code for a *specific release*, for instance, 1.10.0.
+
+We do not force push to any of these branches. If a commit must be reverted, use `git revert`, and push the reverted commit.
+
+Foreman projects strive to have a linear history ([like this](http://i.imgur.com/RMFS3bt.png)). A way to achieve this is to never create merge commits, by always rebasing your changes with develop `git pull --rebase upstream develop`. This allows us to easily revert our code to any point in time, which is helpful to detect in what commit we introduced an error using `git bisect`.
```

Comment:
Personally, I have come to dislike wording like "strives" as it says we think this is the right way but we don't "enforce" it and as a user I would overlook that for the most part. The mention of the 4 projects at the start help to make it clear that this really only applies to those projects. However, I still think it is worth making the distinction and being clear about how core and plugins (even the popular or supported ones) differ in their history and commit management here. Maybe even include mention to the effect of "we encourage developers to adopt this strategy for managing their repositories because we believe that linear history provides X, Y, Z. As well, here are some scripts/tools/strategies on how to use PRs with this merging strategy to get you started."